retrieval/resource_map.go

```diff
@@ -122,6 +149,8 @@ var ResourceMappings = []ResourceMap{
 	},
 	EC2ResourceMap,
 	GCEResourceMap,
+	ProxyResourceMap,
```

Comment:
Do you intend to make the order of `ProxyResourceMap` and `DevappResourceMap` behind  `k8s_container`?

I might be wrong, but I think metrics exposed might be still reported with monitored resource `k8s_*`.

Can you add a test-case and see whether a metric with proxy/devapp monitored resource will be reported as expected when `__meta_kubernetes_{namespace|pod|pod_container|node}_(name)` exist?